### PR TITLE
Sealed Container NPC functionality.

### PR DESCRIPTION
**Pull request by gweivyth**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Adds functionality for Sealed Container NPCs, the start of building out SCNMs.  Kevan, Pogigi, and Lurouillat will accept the appropriate R/EX items and exchange them for the appropriate key items.

Short TODO: Verify showText events are correct after the trade, I used KnowOne's event captures but they might not align with the current retail version anymore as they're pretty dated.  

## Steps to test these changes

Copy and paste lua, talk to NPCs, trade items, etc.

### CODE REVIEW

File: scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua [hunk 1]
```diff
@@ -0,0 +1,45 @@
+-----------------------------------
+-- Area: Bastok Markets [S]
+--  NPC: Kevan
+-- !pos -308.590 -012.000 -094.227 189
+-----------------------------------
+local ID = require("scripts/zones/Bastok_Markets_[S]/IDs")
+require("scripts/globals/keyitems")
+require("scripts/globals/settings")
```
Comment (Xaver-DaRed) [inline scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua]: We don't use keyitems nor settings requires anymore.

File: scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    local itemID = trade:getItemId()
+    local itemKeyItemMapping = {
+        [2686] = xi.ki.THE_WORDS_OF_DONHU_I,
+        [2687] = xi.ki.THE_WORDS_OF_DONHU_II,
+        [2688] = xi.ki.THE_WORDS_OF_DONHU_III,
+        [2689] = xi.ki.THE_WORDS_OF_DONHU_IV,
+        [2690] = xi.ki.THE_WORDS_OF_DONHU_V,
+        [2691] = xi.ki.THE_WORDS_OF_DONHU_VI,
+        [2692] = xi.ki.THE_WORDS_OF_DONHU_VII,
+        [2693] = xi.ki.THE_WORDS_OF_DONHU_VIII
```
Comment (Xaver-DaRed) [inline scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua:22]: This items would regularly need to be added to scripts/globals/items.lua

However, since we are moving that file location, I think we can wait until that gets moved.

File: scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    -- Make sure we're only trading 1 Strongbox at a time.
+    if trade:hasItemQty(itemID, 1) and trade:getItemCount() == 1 and itemKeyItemMapping[itemID] then
```
Comment (Xaver-DaRed) [inline scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua]:
```lua
if
    npcUtil.tradeHasExactly(trade, itemID) and
    itemKeyItemMapping[itemID]
then
```

File: scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        player:tradeComplete()
+        local keyItem = itemKeyItemMapping[itemID]
+        npc:showText(npc, 13549)
+        player:addKeyItem(keyItem)
+        player:messageSpecial(ID.text.KEYITEM_OBTAINED, keyItem)
```
Comment (Xaver-DaRed) [inline scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua]: `npcUtil.giveKeyItem(player, keyItem)` handles the message.
So there would be no need for the message.

Also, we could add the message used in `npc:showText(npc, 13549)` in the IDs file to not use a magic number/raw ID

File: tools/requirements.txt [hunk 5]
```diff
@@ -1,8 +0,0 @@
-mariadb
-gitpython
-pylint
-pyyaml
-colorama
-zmq
-black
-regex
```
Comment (Xaver-DaRed) [inline tools/requirements.txt:8]: Why was this deleted?

File: scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    player:startEvent(334)
+end
+
+entity.onEventUpdate = function(player, csid, option)
+end
+
+entity.onEventFinish = function(player, csid, option)
+end
+
+return entity
```
Comment (Xaver-DaRed) [inline scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua]: Newline at the end of the file needed.

File: scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+-----------------------------------
+local ID = require("scripts/zones/Bastok_Markets_[S]/IDs")
+require("scripts/globals/keyitems")
+require("scripts/globals/settings")
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    local itemID = trade:getItemId()
+    local itemKeyItemMapping = {
```
Comment (Xaver-DaRed) [inline scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua]:
```lua
local itemKeyItemMapping =
{
```

**Comment by Xaver-DaRed** [comment]
Same comments made to the first file apply to the other 2 files.

Also, sup Gweivyth, long time no see.

**Comment by gweivyth** [comment]
>Also, sup Gweivyth, long time no see.

Hi friend!  Decided to start contributing upstream, I wrote a lot of this stuff for Topaz and am just going to convert it over, but I never ended up using it, felt like a waste.  Might do some more stuff afterwards, undecided, still learning the differences in the codebase.  :)

File: scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua (hunk 1 above)
Comment (gweivyth) [reply scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua]: Oh cool!  These were necessary before but I didn't think to try without, thanks!

File: scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua (hunk 4 above)
Comment (gweivyth) [reply scripts/zones/Bastok_Markets_[S]/npcs/Kevan.lua]: Resolved the first half, will resolve second half once I am 100% sure these are even the right events.  I wasn't able to farm any of these items and pull the event ID on retail yet, planning on doing that tomorrow because I don't think the events in the logs I used are still up to date per comment.

File: tools/requirements.txt (hunk 5 above)
Comment (gweivyth) [reply tools/requirements.txt:8]: Cause I wasn't paying attention and it wasn't on ignore.  Will fix. :)

Comment (Xaver-DaRed) [reply tools/requirements.txt:8]: This is still being deleted and it's probably the cause of the CI fail

Comment (gweivyth) [reply tools/requirements.txt:8]: Yeah, not sure if I am gonna be able to fix it retroactively, its on my ignore now but still showing, I'll probably just rebranch and resubmit once I resolve everything here.
